app/models/vendor.rb

```diff
@@ -0,0 +1,47 @@
+class Vendor < ActiveRecord::Base
+  belongs_to :market
+  has_many :sales
+  has_many :products
+
+  validates_presence_of :name, :number_of_employees, :market_id
+  validates_numericality_of :number_of_employees, :market_id
+
+  def recent_sales_first(how_many = sales.count)
+    sorted_sales = sales.sort_by { |sale| sale.purchase_time }
+    sorted_sales.reverse!
```

Comment:
Also, it's generally a bad idea to mutate your data like on line 11 (anything with a `!` after it in Ruby is mutating the data it's being called on). It'd be better to do `reverse_sorted_sales = sorted_sales.reverse`, or even `sorted_sales = sorted_sales.reverse`.